stringdtype/stringdtype/src/dtype.c

```diff
@@ -393,11 +393,10 @@ init_string_dtype(void)
 
     StringDType.singleton = singleton;
 
-    free(StringDType_DTypeSpec.casts[1]->dtypes);
-    free(StringDType_DTypeSpec.casts[1]);
-    free(StringDType_DTypeSpec.casts[2]->dtypes);
-    free(StringDType_DTypeSpec.casts[2]);
-    free(StringDType_DTypeSpec.casts);
+    for (int i = 0; i < 4; i++) {
```

Comment:
`NULL` is the sentinel here, right? We can just iterate until we hit that, so that we don't have to worry about updating the number of casts here.